app/components/verification/ExternalVerifierStatuses.tsx

```diff
@@ -0,0 +1,194 @@
+    ) as Array<[ExternalVerifierKey, ExternalVerifications[ExternalVerifierKey]]>;
+
+    if (verifierEntries.length === 0) {
+      externalVerifierStatusesRef.current = {};
+      setExternalVerifierStatuses({});
+      return;
+    }
+
+    const activeKeys = new Set<ExternalVerifierKey>(verifierEntries.map(([key]) => key));
+    const preservedStatusesEntries = Object.entries(externalVerifierStatusesRef.current).filter(([key]) =>
+      activeKeys.has(key as ExternalVerifierKey)
+    );
+    const preservedStatuses = Object.fromEntries(preservedStatusesEntries);
+    externalVerifierStatusesRef.current = preservedStatuses;
+    setExternalVerifierStatuses(preservedStatuses);
+
+    let isCancelled = false;
+    let timeoutId: number | null = null;
+
+    const shouldFetchKey = (key: ExternalVerifierKey) => {
+      const status = externalVerifierStatusesRef.current[key];
+      if (!status) {
+        return true;
+      }
+      const messageIndicatesPending =
+        typeof status.message === "string" && status.message.toLowerCase().includes("pending");
+      return status.state === "pending" || status.state === "unknown" || messageIndicatesPending;
+    };
+
+    const updateStatuses = async (): Promise<boolean> => {
+      const keysToFetch = verifierEntries.filter(([key]) => shouldFetchKey(key));
+
+      if (keysToFetch.length === 0) {
+        return verifierEntries.some(([key]) => shouldFetchKey(key));
+      }
+
+      const results = await Promise.all(
+        keysToFetch.map(async ([key, data]) => {
+          const status = await requestExternalVerifierStatus(key, data);
+          return [key, status] as const;
+        })
+      );
+
+      if (isCancelled) {
+        return false;
+      }
+
+      const next = { ...externalVerifierStatusesRef.current };
+      results.forEach(([key, status]) => {
+        next[key] = status;
+      });
+
+      externalVerifierStatusesRef.current = next;
+      setExternalVerifierStatuses(next);
+
+      return Object.values(next).some((status) => status.state === "pending" || status.state === "unknown");
+    };
+
+    const pollStatuses = async () => {
+      const hasPending = await updateStatuses();
+      if (isCancelled) return;
+      if (hasPending) {
+        timeoutId = window.setTimeout(pollStatuses, refreshRateSeconds * 1000);
+      }
+    };
+
+    pollStatuses();
+
+    return () => {
+      isCancelled = true;
+      if (timeoutId) {
+        window.clearTimeout(timeoutId);
+      }
+    };
+  }, [verifications, refreshRateSeconds]);
+
+  if (!verifications || !Object.values(verifications).some((value) => !!value)) {
+    return null;
+  }
+
+  return (
+    <div className="bg-white rounded-lg shadow-sm border border-gray-200 p-4 md:p-6 my-6 md:my-8">
+      <div className="flex flex-col md:flex-row md:items-center md:justify-between gap-2 md:gap-4 mb-4">
+        <h2 className="text-lg md:text-xl font-bold text-gray-900">Verification on other verifiers</h2>
```

Comment:
Let's just say "Other Verifiers" in the title and add a small subtitle saying "Sourcify automatically shares contracts with other known verifiers"